src/GraphQL/Reflection/ReflectionExtensions.cs

```diff
@@ -0,0 +1,199 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace GraphQL.Reflection
+{
+    /// <summary>
+    /// Provides extension methods to <see cref="ParameterInfo"/> and <see cref="PropertyInfo"/> instances to read nullable type reference annotations.
+    /// </summary>
+    public static class ReflectionExtensions
+    {
+        /// <summary>
+        /// Examines the parameter type and walks the type definitions, for each returning the nullability information for the type.
+        /// <para>
+        /// For instance, for Test&lt;string, int&gt;, first returns nullability for Test&lt;string, int&gt;, then string, then int.
```

Comment:
It is not clear from this example, in what order you need to traverse the tree of types. You provides a link to nullable-metadata docs so whatever. One can open and read.